src/tuneflow_py/models/song.py

```diff
@@ -83,6 +84,95 @@ def remove_track(self, track_id: str):
                 dep_track.remove_output()
         return track
 
+    def get_structures(self):
+        return self._proto.structures
+
+    def get_structure_at_index(self, index: int):
+        return self._proto.structures[index]
+
+    def get_structure_at_tick(self, tick: int):
+        def tickToStructureFn(tick):
+            return StructureMarker(song=self,tick=tick, type=StructureType.UNKNOWN)._proto
+
+        def structureToTickFn(structure):
+            return structure.tick
+
+        index = Song.get_structure_at_tick_impl(
+            tick, self._proto.structures, tickToStructureFn, structureToTickFn
+        )
+
+        if index < 0:
+            index = 0
+
+        if index >= len(self._proto.structures):
+            index = len(self._proto.structures) - 1
+
+        return self._proto.structures[index]
+
+    @staticmethod
+    def get_structure_at_tick_impl(
```

Comment:
You can combine this function and `get_structure_at_tick` into one method, since there is only one usage. The TS version has separate methods because the impl method was used elsewhere.